### ISSUE TITLE
Fix canonical pretraining startup: LR schedule is referenced before assignment

### ISSUE THREAD

**Issue by mlnomadpy** [mlnomadpy/flaxchat#2]
Parent: #1

Observed on `master` at commit `9653ab4`.

## Problem

In `scripts/pretrain.py`, the optimizer is initialized with:

```python
optimizer = setup_optimizer(..., lr_schedule_fn=lr_schedule)
```

before `lr_schedule` is created by `make_lr_schedule(...)`. The training horizon used to construct the schedule is also resolved only after optimizer creation. As written, the canonical pretraining entry point raises a `NameError` before reaching its first training step.

The script also constructs `wd_schedule` later without clearly wiring it into the optimizer.

## Proposed change

Reorder initialization so that the effective global batch size and training horizon are resolved first, followed by learning-rate and weight-decay schedules, and only then optimizer construction. Store the resolved schedule inputs in the run configuration.

## Acceptance criteria

- A clean CPU smoke configuration starts through the public CLI and completes at least two optimizer steps.
- Unit tests check warmup, peak, decay, and terminal LR values at boundary steps.
- Weight-decay scheduling is either applied consistently or removed; no dead schedule remains.
- The resolved configuration records the effective global batch, total update count, warmup count, and schedule parameters.
- A regression test fails if optimizer construction can reference an uninitialized schedule.

**Comment by mlnomadpy** [comment]
Resolved across the current master series. The public CPU smoke completes two optimizer updates; schedules are constructed only after the effective horizon/batch are resolved; schedule boundary tests and resolved run metadata are covered. Verified by the full local suite (222 passed before the latest additions) and green CPU Actions at e583477.
